daffodil-core/src/main/scala/org/apache/daffodil/dsom/RestrictionUnion.scala

```diff
@@ -76,7 +76,7 @@ final class Restriction(xmlArg: Node, val simpleTypeDef: SimpleTypeDefBase)
     val baseQNameString = baseQNameNodeSeq.text
     val tryBaseQName = QName.resolveRef(baseQNameString, xml.scope,
       tunable.unqualifiedPathStepPolicy)
-    Assert.invariant(tryBaseQName.isSuccess)
+    schemaDefinitionUnless(tryBaseQName.isSuccess, "Invalid base string value.")
```

Comment:
I'm not sure this error message is really clear what the problem is. Also, I think it's possible that resolveRef can fail for reasons other than being invalid, like it just couldn't find the reference. Maybe an error message that might be more clear is something like
```scala
"Failed to resolve base property reference for xs:restriction: " + tryBaseQname.failed.get.getMessage
```
That way it's more clear what "base" is, and it also includes the error message which specifically will say why it failed.